src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py

```diff
@@ -85,10 +86,19 @@ def _run_bulk(self, run: Run, stream=False, **kwargs):
             run._resume_from = self._ensure_run_completed(run._resume_from)
         # Start trace
         if self._config.is_internal_features_enabled():
-            from promptflow.tracing._start_trace import start_trace
+            from promptflow.tracing._start_trace import is_collection_writeable, start_trace
+
+            logger.debug("start trace for flow run...")
+            if is_collection_writeable():
+                logger.debug("trace collection is writeable, will use flow name as collection...")
```

Comment:
what does `writeable` and `protected` mean?

Reply:
`start_trace` receives a parameter `collection`, this is always editable for user, which means everytime user can specify a new one, and we will replace it with the new one.

meanwhile, 1) we have a default value (comes from cwd); 2) for test and batch run, we expect use flow name as collection.

so i regard a user specified collection is protected, so that only another user-specified value can overwrite it; otherwise, it is writeable for 1) and 2).